packages/mds-agency/request-handlers.ts

```diff
@@ -65,7 +86,9 @@ export const registerVehicle = async (req: AgencyApiRequest, res: AgencyApiRespo
 
   const failure = badDevice(device)
   if (failure) {
-    return res.status(400).send(failure)
+    if (typeof failure !== 'boolean') {
+      return res.status(400).send(failure)
+    }
```

Comment:
What's this for?

Reply:
`badDevice` either returns `true` or an error object. The error object is formatted according to the error schema spelled out in the spec. I'll tweak it so it returns `null` if there are no errors, and the `typeof` check can go away.